test/integration/consul-container/test/README.md

```diff
@@ -0,0 +1,97 @@
+# Upgrade Integration Tests
+
+The upgrade integration test is responsible for ensuring that users are able to upgrade between consul versions without experiencing difficulties. The upgrade tests does the following:
+- Ensure that customers are able to upgrade to a new Consul version, and new and existing features work as expected. 
+- This tests aims to ensure a high degree of interoperability between versions of Consul by supporting `n-2` and `n-1` upgrades, ensuring that upgrading to newer versions does not degrade or break functionalities.
+    - Validate users are able to perform a direct upgrade of Consul. For example upgrading from `1.13 → 1.14`.  
+    - Validate users are able to perform skip-version upgrade of Consul. For example upgrading from `1.13 → 1.15`.   
+
+- The upgrade tests also aims to highlight errors that may occur as users attempt to upgrade their current version to a newer version. 
+
+Here is an example of how the upgrade tests work
+1. Create a cluster with a specified number of server and client agents, then enable the feature to be tested. 
+2. Create some workload in the cluster, e.g. register 2 services: static-server, static-client. 
+3. Verify connection / disconnection (e.g. deny Action) 
+4. Upgrade consul cluster and restart the envoy sidecars (we restart envoy sidecar to ensure the upgraded consul binary can read the state from the previous version and generate the correct envoy configurations) 
+5. Verify connection / disconnection (e.g., deny Action)
+
+
+The tests focus on Consul upgrade lifecycle and emit logs to stdout which developers can silence. Please reach out to the [CTIA Team](https://hashicorp.slack.com/app_redirect?channel=team-consul-test-infra) for any questions or help getting started.
```

Comment:
I don't think we need to expose the internal team name and slack link in public repo, so reach out to Consul team should be enough.